staging/src/k8s.io/api/core/v1/types.go

```diff
@@ -2116,6 +2116,18 @@ type Probe struct {
 	// Defaults to 3. Minimum value is 1.
 	// +optional
 	FailureThreshold int32 `json:"failureThreshold,omitempty" protobuf:"varint,6,opt,name=failureThreshold"`
+	// Optional duration in seconds the pod needs to terminate gracefully upon probe failure.
+	// If this value is nil, the default grace period will be used instead. Otherwise, this
+	// value overrides the value provided as part of the pod spec, or the default value if
```

Comment:
The second clause reads wierd.  I would say instead:

```
If this value is nil, the pod's terminationGracePeriodSeconds will be used. Otherwise, this value overrides the value provided by the pod spec.
Value must be a non-negative integer...
```